src/components/learner-credit-management/members-tab/bulk-actions/MemberRemoveModal.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  ActionRow, Alert, Hyperlink, Icon, ModalDialog, Spinner, StatefulButton,
+} from '@edx/paragon';
+import { RemoveCircle } from '@edx/paragon/icons';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { useRequestState } from '../../../subscriptions/licenses/LicenseManagementModals/LicenseManagementModalHook';
+import { configuration } from '../../../../config';
+import LmsApiService from '../../../../data/services/LmsApiService';
+
+/**
+ * Returns StatefulButton labels
+ * @param {number} totalToRemove
+ * @returns {Object}
+ */
+const generateRemoveModalSubmitLabel = (totalToRemove) => {
+  let buttonNumberLabel = 'all';
+
+  if (Number.isFinite(totalToRemove)) {
+    buttonNumberLabel = `(${totalToRemove})`;
+  }
+
+  return {
+    default: `Remove ${buttonNumberLabel}`,
+    pending: `Removing ${buttonNumberLabel}`,
+    complete: 'Done',
+    error: `Retry remove ${buttonNumberLabel}`,
+  };
+};
+
+const MemberRemoveModal = ({
+  isOpen,
+  onClose,
+  onSuccess,
+  usersToRemove,
+  removeAllUsers,
+  totalToRemove,
+  groupUuid,
+}) => {
+  const [requestState, setRequestState, initialRequestState] = useRequestState(isOpen);
+
+  const buttonLabels = generateRemoveModalSubmitLabel(totalToRemove);
+
+  const title = `Remove member${removeAllUsers || totalToRemove > 1 ? 's' : ''}?`;
```

Comment:
We can use [utils.makePlural](https://github.com/openedx/frontend-app-admin-portal/blob/32f8b9956f5353bc6e499c5617d7477afd515699/src/utils.js#L465) here.
```suggestion
  const title = `Remove ${makePlural('member', totalToRemove)}?`;
```

Reply:
No, just because there's also the case of removeAllUsers that we have to account for